CONTRIBUTING.md

```diff
@@ -0,0 +1,187 @@
+# Contributing
+
+Thanks for taking a look at StatsBase!
+
+In general this guide summarizes and extends the [Julia Contributing Guidelines](https://github.com/JuliaLang/julia/blob/master/CONTRIBUTING.md#general-formatting-guidelines-for-julia-code-contributions) and [Julia Style Guide](https://docs.julialang.org/en/latest/manual/style-guide.html).
+
+## Reporting Issues
+
+* It's always helpful to start by reviewing existing issues is and if necessary referencing back to previous discussions.
+* Including minimal sample code (either to reproduce a bug or demonstrating desired behaviour) is greatly appreciated.
+* For bugs, it's helpful to checkout the latest development version (`Pkg.checkout("StatsBase")`) and post the output from `versioninfo()`.
+
+## Pull Requests (PR)s
+
+* Try to keep changes in PRs constrained if possible. This will reduce the amount of work for the reviewer and likely speed up the review process.
+* Using `git diff upstream/master` and `git diff --stat upstream/master` (assuming you've run `git remote add upstream https://github.com/JuliaStats/StatsBase.jl`) can be useful commands for comparing your branch against the upstream master before pushing.
```

Comment:
Well I guess that depends on your workflow, but in either case you'll need to set the remote either for upstream or for your fork. I tend to clone my own fork rather than using `Pkg.checkout` for a few reasons (e.g., I don't accidentally do `git push origin <branch>` on a repo I don't own).